### PR TITLE
Clock module

### PR DESCRIPTION
**Pull request by fabriziovigevani**
**This PR is subject to change depending on the designs**

Includes court module and `heartbeat` interaction.

Made some components used by the `account module` more generic so we could reuse them with the `clock module`.

Will integrate this logic with the balance module to tell if a balance should be updated on a separate PR

**Preview:**

Screen Shot 2020-03-06 at 3 04 57 PM

Screen Shot 2020-03-06 at 3 05 10 PM

### CODE REVIEW

File: src/components/ClockModule.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    setOpened(false)
+  }, [])
+
+  const handleOnClick = useCallback(async () => {
+    try {
+      const tx = await onHeartbeat(neededTransitions)
+      await tx.wait()
+    } catch (err) {
+      console.error(err)
+    }
```
Comment (bpierre) [inline src/components/ClockModule.js]: Note to self: remember to use [`addActivity()`](https://github.com/aragon/court-dashboard/blob/activity/src/components/Activity/ActivityProvider.js#L109) here, whenever #100 or this PR gets merged.

File: src/components/ClockModule.js [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+              width={4.5 * GU}
+              height={4.5 * GU}
+            />
+          </div>
+        }
+        content={
+          <div
+            css={`
+              line-height: 1;
+              width: 120px;
```
Comment (bpierre) [inline src/components/ClockModule.js]:
```suggestion
              width: ${15 * GU}px;
```

File: src/components/ClockModule.js [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+              & > time {
+                & > span:first-child {
+                  display: none;
+                }
```
Comment (bpierre) [inline src/components/ClockModule.js]: This should probably be added as a mode into `Timer`, as the internal HTML structure could break at any time.

File: src/components/ClockModule.js [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+                & * {
+                  line-height: 1;
+                }
```
Comment (bpierre) [inline src/components/ClockModule.js]: Same here, ideally we should either add a mode on the component or try to find another way to achieve this.

File: src/components/ClockModule.js [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+                  `}
+                >
+                  {isSynced ? 'Court term synced' : 'Term out of sync'}
+                </span>
+              </>
+            ) : (
+              <div
+                css={`
+                  background: ${theme.background};
+                  height: 40px;
```
Comment (bpierre) [inline src/components/ClockModule.js]:
```suggestion
                  height: ${5 * GU}px;
```

File: src/components/ClockModule.js [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+                      ? theme.contentSecondary
+                      : theme.negative};
+                    line-height: 1;
+                  `}
+                >
+                  {isSynced ? 'Court term synced' : 'Term out of sync'}
+                </span>
+              </>
+            ) : (
+              <div
```
Comment (bpierre) [inline src/components/ClockModule.js]: Nice touch! :heart:

File: src/components/ClockModule.js [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+              />
+            )}
+          </div>
+        }
+        onClick={toggle}
+      />
+
+      <HeaderPopover
+        animateHeight={false}
+        heading="Court cLock"
```
Comment (bpierre) [inline src/components/ClockModule.js]:
```suggestion
        heading="Court clock"
```

File: src/components/ClockModule.js [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+          >
+            {isSynced ? (
+              <span
+                css={`
+                  ${textStyle('body3')};
+                  color: ${theme.contentSecondary};
+                `}
+              >
+                Term is the time unit for measuring Aragon Court phases. Each
+                term lasts {formatDuration(courtConfig.termDuration / 1000)}
```
Comment (bpierre) [inline src/components/ClockModule.js]:
```suggestion
                term lasts {formatDuration(courtConfig.termDuration / 1000)}.
```

File: src/components/ClockModule.js [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+              </span>
+            ) : (
+              <Button
+                label="Call Heartbeat"
```
Comment (bpierre) [inline src/components/ClockModule.js]: Not sure if “heartbeat” is a name or a noun here, but if it’s the latter we shouldn’t capitalize it:

```suggestion
                label="Call heartbeat"
```

File: src/components/Header/Header.js [hunk 10]
```diff
@@ -30,13 +30,16 @@ const Header = React.memo(function Header({ compactMode, toggleMenuPanel }) {
       <div
         css={`
           display: flex;
-          flex-grow: 1;
-          justify-content: center;
+          align-self: stretch;
+
+          & > * {
+            align-self: stretch;
+          }
```
Comment (bpierre) [inline src/components/Header/Header.js]: Is this needed?

File: src/components/Header/Header.js [hunk 11]
```diff
@@ -30,13 +30,16 @@ const Header = React.memo(function Header({ compactMode, toggleMenuPanel }) {
...
         `}
       >
-        {!compactMode && <Clock />}
+        {!compactMode && <ClockModule />}
+        <Account />
```
Comment (bpierre) [inline src/components/Header/Header.js]: Should we rename this `AccountModule` for consistency?

Comment (fabriziovigevani) [reply src/components/Header/Header.js]: Yep good idea

File: src/components/ClockModule.js (hunk 9 above)
Comment (fabriziovigevani) [reply src/components/ClockModule.js]: I think in this case is intended to be a name? cc. @dizzypaty

File: src/components/Header/Header.js (hunk 10 above)
Comment (fabriziovigevani) [reply src/components/Header/Header.js]: Not at all, forgot to remove it when updating the inner components. The `align-self` property on the parent component can be replaced with `height: 100%`

File: src/components/ClockModule.js (hunk 3 above)
Comment (fabriziovigevani) [reply src/components/ClockModule.js]: Yeah that's true! I can add the mode in `@aragon/ui` and i'll add a note here to use it when its available :)

Comment (fabriziovigevani) [reply src/components/ClockModule.js]: PR: https://github.com/aragon/aragon-ui/pull/734

File: src/components/ClockModule.js (hunk 1 above)
Comment (fabriziovigevani) [reply src/components/ClockModule.js]: Added in https://github.com/aragon/court-dashboard/pull/227/commits/f1b5e8d2f1ef093f28f39f709b132f82f3a1851a

File: src/components/ClockModule.js (hunk 9 above)
Comment (fabriziovigevani) [reply src/components/ClockModule.js]: From an offline chat with @dizzypaty, we are rewording most labels including the `Call heartbeat ` to `Update term`. Here is the commit https://github.com/aragon/court-dashboard/pull/227/commits/e80c35b1e5454cfa489dceaba8fdb4d31a08b08e

File: src/components/Activity/activity-types.js [hunk 12]
```diff
@@ -101,12 +102,12 @@ const ACTIVITY_TYPES = new Map(
         `,
       }
     },
-    revealVote({ roundId, disputeId }) {
+    heartbeat({ transitions }) {
       return {
-        title: 'Reveal vote',
-        icon: iconAnj,
+        title: 'Heartbeat',
```
Comment (bpierre) [inline src/components/Activity/activity-types.js]: Shouldn’t this be “Update term”, based on https://github.com/aragon/court-dashboard/pull/227#discussion_r390451950?

Comment (fabriziovigevani) [reply src/components/Activity/activity-types.js]: Yeah totally! Good catch!

File: src/components/ClockModule.js [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+        outline: 0;
+      `}
+    >
+      <HeaderModule
+        icon={
+          <div
+            css={`
+              background-color: ${theme.background};
+              display: flex;
+              border-radius: 100px;
```
Comment (sohkai) [inline src/components/ClockModule.js:78]: Wondering: are we just trying to turn this into a circle? We should use `border-radius: 50%` instead if so :)

File: src/components/ClockModule.js [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+          <div
+            css={`
+              line-height: 1;
+              width: ${15 * GU}px;
+
+              & span {
+                line-height: 1;
+              }
+
+              // TODO: Use showIcon new prop when available, see: https://github.com/aragon/aragon-ui/pull/734
```
Comment (sohkai) [inline src/components/ClockModule.js:99]: cc @bpierre have we released aragonUI since this was introduced?

Comment (bpierre) [reply src/components/ClockModule.js:99]: Nope, I’m going to update this after the new version gets released. :+1:

**Pull request merged**